pkgs/development/python-modules/netcdf4/default.nix

```diff
@@ -1,50 +1,73 @@
 { stdenv, buildPythonPackage, fetchPypi, isPyPy, pytest
-, numpy, zlib, netcdf, hdf5, curl, libjpeg, cython, cftime
-}:
-buildPythonPackage rec {
-  pname = "netCDF4";
-  version = "1.5.1.2";
+, numpy, zlib, netcdf, hdf5, curl, libjpeg, cython, cftime 
+,mpi4py ? null, openssh ? null}:
+assert (mpi4py!= null) -> 
+  openssh != null
+  && hdf5.mpi == mpi4py.mpi 
+  && netcdf.mpi == mpi4py.mpi;
 
-  disabled = isPyPy;
+let 
+  mpiSupport = (mpi4py!=null);
+  mpi= if mpiSupport then mpi4py.mpi else null;
+in
+  buildPythonPackage rec {
+    pname = "netCDF4";
+    version = "1.5.1.2";
+  
+    disabled = isPyPy;
+    
+    src = fetchPypi {
+      inherit pname version;
+      sha256 = "161pqb7xc9nj0dlnp6ply8c6zv68y1frq619xqfrpmc9s1932jzk";
+    };
+  
+    checkInputs = [ pytest ];
+  
+    buildInputs = [
+      cython 
+      mpi 
+    ];
+  
+    propagatedBuildInputs = [
+      cftime
+      numpy
+      zlib
+      netcdf
+      hdf5
+      curl
+      libjpeg
+    ] ++ (if mpiSupport then [mpi4py openssh] else []);
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "161pqb7xc9nj0dlnp6ply8c6zv68y1frq619xqfrpmc9s1932jzk";
-  };
+    patches=[./parallel4Detection.patch];
+  
+    preBuild= if mpiSupport then ''
```

Comment:
`makeFlagsArray` can be set directly. Use `lib.optionals` to extend when `mpiSupport`.

Reply:
I would like to address this but I am not sure how (I do not know what is in the makeFlagsArray originally I was just happy to add the required lines)
I would like to know what I should have done to learn.

Reply:
```suggestion
    makeFlags = lib.optionals mpiSupport [ "CC=mpicc" "CXX=mpicxx" ];
```